arangod/RestServer/Metric/Metric.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2021 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Kaveh Vahedipour
+////////////////////////////////////////////////////////////////////////////////
+
+#include <RestServer/counter.h>
+
+#include <ostream>
+#include <string_view>
+
+namespace arangodb::metric {
+
+class Metric {
+ public:
+  Metric(std::string_view name, std::string_view help, std::string_view labels) noexcept;
+
+  [[nodiscard]] std::string_view help() const noexcept;
+  [[nodiscard]] std::string_view name() const noexcept;
+  [[nodiscard]] std::string_view labels() const noexcept;
+
+  virtual std::string_view type() const noexcept = 0;
+  virtual void toPrometheus(std::string& result, std::string_view globals,
+                            std::string_view alternativeName) const = 0;
+  virtual void toPrometheusBegin(std::string& result, std::string_view name) const;
+  virtual ~Metric();
+
+ private:
+  std::string_view _name;
```

Comment:
Do we currently have any metrics with dynamic names? If yes, then we will likely need an `std::string` here to store the name. At least I guess that dynamic names are generated at call sites that generate the `Metric` objects and probably go out of scope while the `Metric` object lives on afterwards.

Reply:
> Do we currently have any metrics with dynamic names?
No
Their name is `#name` in macros

Reply:
I was rather thinking about metrics that will be created only at runtime, e.g. metrics for certain views or collections (which could come and go while the server is running). If we don't have such metrics right now and all names are known at compile-time, then it is fine as it is.

Reply:
> If we don't have such metrics right now and all names are known at compile-time, then it is fine as it is.

We have, arango search metrics, but they have only dynamic labels, not names. You cant create metric with dynamic name now